packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -183,11 +204,53 @@ export class HttpRoute extends Resource implements IHttpRoute {
       routeKey: props.routeKey.key,
       target: `integrations/${integration.integrationId}`,
       authorizerId: authBindResult?.authorizerId,
-      authorizationType: authBindResult?.authorizationType ?? 'NONE', // must be explicitly NONE (not undefined) for stack updates to work correctly
+      authorizationType: Lazy.string({
+        produce: () => {
+          if (this.iamEnabled) {
+            return 'AWS_IAM';
+          } else {
+            return authBindResult?.authorizationType ?? 'NONE'; // must be explicitly NONE (not undefined) for stack updates to work correctly
+          }
+        },
+      }),
       authorizationScopes,
     };
 
     const route = new CfnRoute(this, 'Resource', routeProps);
     this.routeId = route.ref;
   }
+
+  /**
+   * Enable IAM authorization on the route.
+   */
+  public enableIamAuthorization(): void {
+    if (this.authorizer) {
+      throw new Error('authorizationType has been set, so we cannot enable AWS_IAM authorization');
+    }
+
+    this.iamEnabled = true;
+  }
+
+  public grantInvoke(grantee: iam.IGrantable, options: GrantInvokeOptions = {}): iam.Grant {
+    this.enableIamAuthorization();
+
+    const httpMethod = options.httpMethod ?? HttpMethod.ANY;
+    const stage = '*';
```

Comment:
I couldn't find where to get the `stage` variable, so I opted for a wildcard for this part of the resource ARN.